base/linalg/bidiag.jl

```diff
@@ -4,121 +4,74 @@
 mutable struct Bidiagonal{T} <: AbstractMatrix{T}
     dv::Vector{T} # diagonal
     ev::Vector{T} # sub/super diagonal
-    isupper::Bool # is upper bidiagonal (true) or lower (false)
-    function Bidiagonal{T}(dv::Vector{T}, ev::Vector{T}, isupper::Bool) where T
+    uplo::Char    # upper bidiagonal ('U') or lower ('L')
+    function Bidiagonal{T}(dv::Vector{T}, ev::Vector{T}, uplo::Char) where T
         if length(ev) != length(dv)-1
             throw(DimensionMismatch("length of diagonal vector is $(length(dv)), length of off-diagonal vector is $(length(ev))"))
         end
-        new(dv, ev, isupper)
+        new(dv, ev, uplo)
+    end
+    function Bidiagonal(dv::Vector{T}, ev::Vector{T}, uplo::Char) where T
+        Bidiagonal{T}(dv, ev, uplo)
     end
 end
+
 """
-    Bidiagonal(dv, ev, isupper::Bool)
+    Bidiagonal(dv, ev, uplo::Symbol)
 
-Constructs an upper (`isupper=true`) or lower (`isupper=false`) bidiagonal matrix using the
-given diagonal (`dv`) and off-diagonal (`ev`) vectors.  The result is of type `Bidiagonal`
+Constructs an upper (`uplo=:U`) or lower (`uplo=:L`) bidiagonal matrix using the
+given diagonal (`dv`) and off-diagonal (`ev`) vectors. The result is of type `Bidiagonal`
 and provides efficient specialized linear solvers, but may be converted into a regular
-matrix with [`convert(Array, _)`](@ref) (or `Array(_)` for short). `ev`'s length
+matrix with [`convert(Array, _)`](@ref) (or `Array(_)` for short). The length of `ev`
 must be one less than the length of `dv`.
 
-# Example
-
+# Examples
 ```jldoctest
-julia> dv = [1; 2; 3; 4]
+julia> dv = [1, 2, 3, 4]
 4-element Array{Int64,1}:
  1
  2
  3
  4
 
-julia> ev = [7; 8; 9]
+julia> ev = [7, 8, 9]
 3-element Array{Int64,1}:
  7
  8
  9
 
-julia> Bu = Bidiagonal(dv, ev, true) # ev is on the first superdiagonal
+julia> Bu = Bidiagonal(dv, ev, :U) # ev is on the first superdiagonal
 4×4 Bidiagonal{Int64}:
  1  7  ⋅  ⋅
  ⋅  2  8  ⋅
  ⋅  ⋅  3  9
  ⋅  ⋅  ⋅  4
 
-julia> Bl = Bidiagonal(dv, ev, false) # ev is on the first subdiagonal
+julia> Bl = Bidiagonal(dv, ev, :L) # ev is on the first subdiagonal
 4×4 Bidiagonal{Int64}:
  1  ⋅  ⋅  ⋅
  7  2  ⋅  ⋅
  ⋅  8  3  ⋅
  ⋅  ⋅  9  4
 ```
 """
-Bidiagonal(dv::AbstractVector{T}, ev::AbstractVector{T}, isupper::Bool) where {T} = Bidiagonal{T}(collect(dv), collect(ev), isupper)
+function Bidiagonal(dv::AbstractVector{T}, ev::AbstractVector{T}, uplo::Symbol) where T
+    Bidiagonal{T}(collect(dv), collect(ev), char_uplo(uplo))
+end
 Bidiagonal(dv::AbstractVector, ev::AbstractVector) = throw(ArgumentError("did you want an upper or lower Bidiagonal? Try again with an additional true (upper) or false (lower) argument."))
```

Comment:
Woops, missed to update this one. Can we just remove this method? IMO a `MethodError` is better:

```julia
julia> Bidiagonal(a, b)
ERROR: MethodError: no method matching Bidiagonal(::Array{Float64,1}, ::Array{Float64,1})
Closest candidates are:
  Bidiagonal(::Array{T,1}, ::Array{T,1}, ::Char) where T at linalg/bidiag.jl:15
  Bidiagonal(::AbstractArray{T,1}, ::AbstractArray{T,1}, ::Symbol) where T at linalg/bidiag.jl:59
  Bidiagonal(::AbstractArray{Td,1}, ::AbstractArray{Te,1}, ::Symbol) where {Td, Te} at linalg/bidiag.jl:64
```